examples/PhasorDynamics/Example1/example1.cpp

```diff
@@ -71,86 +71,96 @@ int main()
 
   double dt = 1.0 / 4.0 / 60.0;
 
-  std::stringstream buffer;
+  // A data structure to keep track of the data we want to
+  // compare to the reference solution. Rather than keeping
+  // the entire solution vector at every time step around,
+  // we instead narrow down exactly what we want to keep.
+  struct OutputData
+  {
+    double ti, Vr, Vi, dw;
+  };
+
+  // A list of output for each time step.
+  std::vector<OutputData> output;
+
+  // A callback which will be called by the integrator after
+  // each time step. It will be told the time of the current
+  // state, and it is allowed to access the up-to-date state
+  // of the components, which are captured by a closure
+  // due to the [&] notation (every variable that is referenced
+  // by the callback that is external to the callback itself -
+  // here output, bus1, and gen - will be considered a
+  // reference to that variable inside the callback). We select
+  // the subset of the output we're interested in recording and
+  // push it into output, which is updated outside the callback.
+  auto output_cb = [&](double t)
+  {
+    output.push_back({.ti = t,
+                      .Vr = bus1.Vr(),
+                      .Vi = bus1.Vi(),
+                      .dw = gen.y()[1]});
+  };
+
+  // The above lambda is equivalent to writing
+  // struct
+  // {
+  //   std::vector<OutputData>& output;
+  //   Bus<double, size_t>&     bus1;
+  //   Genrou<double, size_t>&  gen;
+
+  //   void operator()(double t)
+  //   {
+  //     output.push_back({.ti = t,
+  //                       .Vr = bus1.Vr(),
+  //                       .Vi = bus1.Vi(),
+  //                       .dw = gen.y()[1]});
+  //   }
+  // } output_cb = {output, bus1, gen};
```

Comment:
I would think the lambda is equivalent to
```c++
struct OutData
{
  real_type ti;
  real_type Vr;
  real_type Vi;
  real_type dw;
};

std::vector<OutData> output;
OutData tmp;

tmp.ti = t;
tmp.Vr = bus.Vr;
tmp.Vi = bus.Vi;
tmp.dw = gen.y()[1];
output.push_back(tmp);
```
I think this description is a little easier to follow. In both cases, it is assumed `t`, `bus1`, `bus2`, and `gen` are instantiated and their values set.

Reply:
No, the struct version is literally what the compiler constructs when it encounters the lambda. This makes it clear that the versions of `bus`, `gen`, and `output` that are used in the lambda are actually *references* to the external variables.